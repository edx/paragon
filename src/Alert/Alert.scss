@import "variables";

//
// Override Bootstrap's alert definition.
// We do this to get more control over colors
// used in alerts.
//

//
// Base styles
//

.alert {
  position: relative;
  padding: $alert-padding-y $alert-padding-x;
  margin-bottom: $alert-margin-bottom;
  border: $alert-border-width solid transparent;
  @include border-radius($alert-border-radius);
}

// Headings for larger alerts
.alert-heading {
  // Specified to prevent conflicts of changing $headings-color
  color: inherit;
}

// Provide class for links that match alerts
.alert-link {
  font-weight: $alert-link-font-weight;
}


// Dismissible alerts
//
// Expand the right padding and account for the close button's positioning.

// Baking in $close-font-size: $font-size-base * 1.5 !default; to avoid any dependency
.alert-dismissible {
  padding-right: ($font-size-base * 1.5) + $alert-padding-x * 2;

  // Adjust close link position
  .close {
    position: absolute;
    top: 0;
    right: 0;
    padding: $alert-padding-y $alert-padding-x;
    color: inherit;
  }
}


// Alternate styles
//
// Generate contextual modifier classes for coloring the alert.

@each $color, $value in $theme-colors {
  .alert-#{$color} {
    color: inherit;
    background-color: theme-color($color, "background");
    border-color: theme-color($color, "border");
  }
}
