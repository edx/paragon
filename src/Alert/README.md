---
title: 'Alert'
type: 'component'
status: 'Stable'
designStatus: 'Done'
devStatus: 'Done'
notes: |

---

# Alert


<p style={{ fontSize: '1.5em' }}>
  This is a pass through component from React-Bootstrap.<br/>
  <a href="https://react-bootstrap.github.io/components/alerts/" target="_blank" rel="noopener noreferrer">
    See React-Bootstrap for documentation.
  </a>
</p>

##### Basic Usage

```jsx live
<>
<Alert variant="success">
  This is a "success" alert with{' '}
  <Alert.Link href="#">an example link</Alert.Link>. Give it a click if you
  like.
</Alert>
<Alert variant="info">
  This is a "info" alert with{' '}
  <Alert.Link href="#">an example link</Alert.Link>. Give it a click if you
  like.
</Alert>
<Alert variant="danger">
  This is a "danger" alert with{' '}
  <Alert.Link href="#">an example link</Alert.Link>. Give it a click if you
  like.
</Alert>
<Alert variant="warning">
  This is a "warning" alert with{' '}
  <Alert.Link href="#">an example link</Alert.Link>. Give it a click if you
  like.
</Alert>
</>
```

##### Kitchen Sink

```jsx live
<Alert variant="success" dismissible>
  <Alert.Heading>Hey, nice to see you</Alert.Heading>
  <p>
    Aww yeah, you successfully read this important alert message. This example
    text is going to run a bit longer so that you can see how spacing within an
    alert works with this kind of content.
  </p>
  <hr />
  <p className="mb-0">
    Whenever you need to, be sure to use margin utilities to keep things nice
    and tidy.
  </p>
</Alert>
```
