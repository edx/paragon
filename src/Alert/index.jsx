export { default } from 'react-bootstrap/Alert';
