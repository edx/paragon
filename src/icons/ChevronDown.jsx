import React from 'react';
import PropTypes from 'prop-types';

const ChevronDown = ({ height, ...other }) => (
  <svg
    height={height}
    viewBox="0 0 24 24"
    version="1.1"
    xmlns="http://www.w3.org/2000/svg"
    {...other}
  >
    <polygon points="12 12.8284271 16.2426407 8.58578644 17.6568542 10 12 15.6568542 6.34314575 10 7.75735931 8.58578644 12 12.8284271" />
  </svg>
);

ChevronDown.propTypes = {
  height: PropTypes.string,
};

ChevronDown.defaultProps = {
  height: '2rem',
};

export default ChevronDown;
