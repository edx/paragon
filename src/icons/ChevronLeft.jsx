import React from 'react';
import PropTypes from 'prop-types';

const ChevronLeft = ({ height, ...other }) => (
  <svg
    height={height}
    viewBox="0 0 24 24"
    version="1.1"
    xmlns="http://www.w3.org/2000/svg"
    {...other}
  >
    <polygon points="11.1715729 12 15.4142136 16.2426407 14 17.6568542 8.34314575 12 14 6.34314575 15.4142136 7.75735931 11.1715729 12" />
  </svg>
);

ChevronLeft.propTypes = {
  height: PropTypes.string,
};

ChevronLeft.defaultProps = {
  height: '2rem',
};

export default ChevronLeft;
