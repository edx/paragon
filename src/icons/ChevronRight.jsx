import React from 'react';
import PropTypes from 'prop-types';

const ChevronRight = ({ height, width, ...other }) => (
  <svg
    height={height}
    width={width}
    viewBox="0 0 24 24"
    {...other}
  >
    <polygon points="12.8284271 12 8.58578644 7.75735931 10 6.34314575 15.6568542 12 10 17.6568542 8.58578644 16.2426407 12.8284271 12" />
  </svg>
);

ChevronRight.propTypes = {
  height: PropTypes.string,
  width: PropTypes.string,
};

ChevronRight.defaultProps = {
  height: '1.5rem',
  width: '1.5rem',
};

export default ChevronRight;
