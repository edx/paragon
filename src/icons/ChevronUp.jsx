import React from 'react';
import PropTypes from 'prop-types';

const ChevronUp = ({ height, width, ...other }) => (
  <svg
    height={height}
    width={width}
    viewBox="0 0 24 24"
    {...other}
  >
    <polygon points="12 11.1715729 7.75735931 15.4142136 6.34314575 14 12 8.34314575 17.6568542 14 16.2426407 15.4142136" />
  </svg>
);

ChevronUp.propTypes = {
  height: PropTypes.string,
  width: PropTypes.string,
};

ChevronUp.defaultProps = {
  height: '1.5rem',
  width: '1.5rem',
};

export default ChevronUp;
