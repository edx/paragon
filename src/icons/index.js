export { default as ChevronDown } from './ChevronDown';
export { default as ChevronUp } from './ChevronUp';
export { default as ChevronRight } from './ChevronRight';
export { default as ChevronLeft } from './ChevronLeft';
export { default as Plus } from './Plus';
export { default as PlusCircle } from './PlusCircle';
export { default as Minus } from './Minus';
export { default as MinusCircle } from './MinusCircle';
export { default as Times } from './Times';
export { default as TimesCircle } from './TimesCircle';
