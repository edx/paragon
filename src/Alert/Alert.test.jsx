import React from 'react';
import renderer from 'react-test-renderer';

import Alert from './index';

import { Info } from '../../icons';

describe('<Alert />', () => {
  describe('correct rendering', () => {
    it('renders without icon prop', () => {
      const tree = renderer.create((
        <Alert>Alert</Alert>
      )).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('renders with correct with icon', () => {
      const tree = renderer.create((
        <Alert icon={Info}>Alert</Alert>
      )).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
