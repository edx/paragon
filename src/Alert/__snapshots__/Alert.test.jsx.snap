// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> correct rendering renders with correct with icon 1`] = `
<div
  className="fade alert-content undefined alert show"
  role="alert"
>
  <span
    className="pgn__icon alert-icon"
  >
    <svg
      aria-hidden={true}
      aria-label=""
      fill="none"
      focusable={false}
      height={24}
      role="img"
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
        fill="currentColor"
      />
    </svg>
  </span>
  <div
    className="alert-message-content"
  >
    Alert
  </div>
</div>
`;

exports[`<Alert /> correct rendering renders without icon prop 1`] = `
<div
  className="fade alert-content undefined alert show"
  role="alert"
>
  <div
    className="alert-message-content"
  >
    Alert
  </div>
</div>
`;
